src/transformers/generation/beam_search.py

```diff
@@ -194,13 +196,6 @@ def __init__(
                 f" divisible by `num_beam_groups`, but is {num_beam_groups} with `num_beams` being {num_beams}."
             )
 
-        if "max_length" in kwargs:
```

Comment:
`max_length` is now a mandatory argument, so this warning no longer makes sense. The test that confirms that this warning is thrown was also removed.

Reply:
Think we can delete `kwargs` then also from the `__init__`